internal/resources/kustomization/resource_kustomization.go

```diff
@@ -40,3 +55,427 @@ var kustomizationSchema = map[string]*schema.Schema{
+	if err != nil {
+		if !clienterrors.IsNotFoundError(err) {
+			return diag.FromErr(err)
+		}
+
+		err = enableContinuousDelivery(&config, scopedFullnameData, common.ConstructMeta(d))
+		if err != nil {
+			return diag.FromErr(errors.Wrapf(err, "Unable to create Tanzu Mission Control kustomization entry, name : %s", kustomizationName))
+		}
+
+		return resourceKustomizationCreate(ctx, d, m)
+	}
+
+	var updateAvailable bool
+
+	if updateCheckForMeta(d, meta) {
+		updateAvailable = true
+	}
+
+	if updateCheckForSpec(d, atomicSpec, scopedFullnameData.Scope) {
+		updateAvailable = true
+	}
+
+	if !updateAvailable {
+		log.Printf("[INFO] kustomization update is not required")
+		return
+	}
+
+	switch scopedFullnameData.Scope {
+	case commonscope.ClusterScope:
+		if scopedFullnameData.FullnameCluster != nil {
+			kustomizationReq := &kustomizationclustermodel.VmwareTanzuManageV1alpha1ClusterNamespaceFluxcdKustomizationKustomizationRequest{
+				Kustomization: &kustomizationclustermodel.VmwareTanzuManageV1alpha1ClusterNamespaceFluxcdKustomizationKustomization{
+					FullName: scopedFullnameData.FullnameCluster,
+					Meta:     meta,
+					Spec:     atomicSpec,
+				},
+			}
+
+			_, err = config.TMCConnection.ClusterKustomizationResourceService.VmwareTanzuManageV1alpha1ClusterFluxcdKustomizationResourceServiceUpdate(kustomizationReq)
+			if err != nil && !clienterrors.IsFeatureDisabledError(err) {
+				return diag.FromErr(errors.Wrapf(err, "Unable to update Tanzu Mission Control cluster kustomization entry, name : %s", kustomizationName))
+			}
+		}
+	case commonscope.ClusterGroupScope:
+		if scopedFullnameData.FullnameClusterGroup != nil {
+			kustomizationReq := &kustomizationclustergroupmodel.VmwareTanzuManageV1alpha1ClustergroupNamespaceFluxcdKustomizationKustomizationRequest{
+				Kustomization: &kustomizationclustergroupmodel.VmwareTanzuManageV1alpha1ClustergroupNamespaceFluxcdKustomizationKustomization{
+					FullName: scopedFullnameData.FullnameClusterGroup,
+					Meta:     meta,
+					Spec: &kustomizationclustergroupmodel.VmwareTanzuManageV1alpha1ClustergroupNamespaceFluxcdKustomizationSpec{
+						AtomicSpec: atomicSpec,
+					},
+				},
+			}
+
+			_, err = config.TMCConnection.ClusterGroupKustomizationResourceService.VmwareTanzuManageV1alpha1ClustergroupFluxcdKustomizationResourceServiceUpdate(kustomizationReq)
+			if err != nil && !clienterrors.IsFeatureDisabledError(err) {
+				return diag.FromErr(errors.Wrapf(err, "Unable to update Tanzu Mission Control cluster group kustomization entry, name : %s", kustomizationName))
+			}
+		}
+	case commonscope.UnknownScope:
+		return diag.Errorf("no valid scope type block found: minimum one valid scope type block is required among: %v. Please check the schema.", strings.Join(scope.ScopesAllowed[:], `, `))
+	}
+
+	if err != nil {
+		err = enableContinuousDelivery(&config, scopedFullnameData, common.ConstructMeta(d))
```

Comment:
check the need of this block. same comment as above for a similar block

Reply:
Updated.